### DISCUSSION TITLE
Do you know whether winmerge is a structured code merging tool

### DISCUSSION THREAD

**Discussion by cena1001** [WinMerge/winmerge#1743] [Q&A] [answered]
Is there any papers related to the algorithm theory used by this software?Is it a structured code merging tool? What is the specific principle？

**Comment by sdottaka** [comment, marked as answer]
Unfortunately, WinMerge is not a structured code merging tool.
For other structured code diff tools and papers, the following URLs may be helpful.

https://difftastic.wilfred.me.uk/tree_diffing.html
https://github.com/Wilfred/difftastic
https://github.com/xaizek/zograscope

**Reply by cena1001** [reply]
thanks~
